### PR TITLE
ISSUE-571 JMAP CalendarEvent/parse for organizer, participants, extensionFields

### PR DESCRIPTION
**Pull request by vttranlina**
resolve https://github.com/linagora/tmail-backend/issues/571

### CODE REVIEW

**Review by Arsnael — COMMENTED** [review]
Read it

**Review by chibenwa — APPROVED** [review]

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala [hunk 1]
```diff
@@ -154,21 +182,76 @@ trait LinagoraCalendarEventParseMethodContract {
...
+           |                    "name": "Raphael OUAZANA",
+           |                    "mailto": "ouazana@domain.tld"
+           |                },
+           |                "participants": [
+           |                    {
+           |                        "name": "Matthieu EXT_BAECHLER",
+           |                        "mailto": "baechler@domain.tld",
+           |                        "kind": "INDIVIDUAL",
+           |                        "role": "ADMIN",
+           |                        "participationStatus": "NEEDS-ACTION",
```
Comment (quantranhong1999) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala:200]: Should we lowercase these fields?

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala [hunk 2]
```diff
@@ -154,21 +182,76 @@ trait LinagoraCalendarEventParseMethodContract {
...
+           |                        "kind": "INDIVIDUAL",
+           |                        "participationStatus": "NEEDS-ACTION",
+           |                        "expectReply": true
+           |                    }
+           |                ],
+           |                "extensionFields": {
+           |                    "X-OPENPAAS-TEST": [
+           |                        "test1",
+           |                        "test2"
+           |                    ],
```
Comment (quantranhong1999) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala:250]: Maybe update the new format in our extension specs?

**Review by quantranhong1999 — APPROVED** [review]

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala (hunk 1 above)
Comment (vttranlina) [reply tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala:200]: The spec is not clear should be lower or upper, so we should keep it original

**Review by quantranhong1999 — APPROVED** [review]

**Comment by vttranlina** [comment]
rebase

**Pull request merged**